components/results_table/commons/results_table.lua

```diff
@@ -293,6 +258,47 @@ function BaseResultsTable:buildNonTeamOpponentConditions()
 	return opponentConditions
 end
 
+-- todo: adjust once #1802 is done
+function BaseResultsTable:buildTeamOpponentConditions()
+	local config = self.config
+
+	local opponent = Team.queryDB('teampage', config.opponent)
+	if not opponent then
+		error('Missing team template for team: ' .. opponent)
+	end
+
+	local opponentPageNames
+	if config.resolveOpponent then
+		opponent = Team.queryDB('teampage', mw.ext.TeamLiquidIntegration.resolve_redirect(opponent))
+		if not opponent then
+			error('Missing team template for team: ' .. opponent)
+		end
+		opponentPageNames = {opponent}
+	elseif Team.queryHistoricalNames then
+		opponentPageNames = Team.queryHistoricalNames(opponent)
+	else
+		opponentPageNames = {opponent}
+	end
+
+	if config.playerResultsOfTeam then
+		return self:buildPlayersOnTeamOpponentConditions(opponentPageNames)
+	end
+
+	local opponentConditions = ConditionTree(BooleanOperator.any)
+
+	for _, pageName in pairs(opponentPageNames) do
+		opponentConditions:add{
+			ConditionNode(ColumnName('opponentname'), Comparator.eq, pageName),
+			ConditionNode(ColumnName('opponentname'), Comparator.eq, pageName:gsub(' ', '_')),
+		}
+	end
```

Comment:
Wouldn't it be better to query on `opponenttemplate` instead of `opponentname`?

Reply:
the current versions use teams pagenames
we can switch it but we would need to add a function for fetching the sub tts from a given tt
`Team.queryHistoricalNames` only fetches the pagenames

Reply:
Very small change required, I think I've already written such a function somewhere. brb

Reply:
`Array.extractValues(Team.queryHistorical(...))`

Reply:
hmm the function (`buildTeamOpponentConditions`) would need several more changes

will play around with it a bit

Reply:
reminder for myself: pXtemplate is resolved --> change accordingly